internal/compiler/widgets/material-base/button.slint

```diff
@@ -56,11 +58,13 @@ export component Button {
         if (root.icon.width > 0 && root.icon.height > 0): Image {
             source <=> root.icon;
             width: 24px;
-            opacity: i-text.opacity;
+            opacity: root.text-opacity;
         }
 
-        i-text := Text {
-            color: root.primary ? Palette.on-primary : Palette.on-secondary-container;
+        if (root.text != "" || root.icon.width <= 0 || root.icon.height <= 0) : Text {
```

Comment:
I think `if (root.text != "")` should be enough. Also for fluent.

Reply:
@Guiguiprim thank you for the PR. 

I looks like the binding `opacity: root.text-opacity;` on `Text` causes the trouble. Without it the text is displayed. Maybe it's a problem with the binding @ogoffart?

Reply:
@Guiguiprim can you add an icon only example to the gallery `controls_page` on the `Buttons` `GroupBox`, please.

Reply:
For the material one I need to add a `min-width` to the root item otherwise the button has a width of zero if neither an icon nor a text is given. The fluent one already has one and is OK.

Reply:
Ok  I see. Can you add please `min-width: max(40px, i-layout.min-width)` to the material button?

Reply:
Done and done

Reply:
Thank you.